include/seqan3/alignment/matrix/alignment_matrix_formatter.hpp

```diff
@@ -334,59 +331,68 @@ struct alignment_matrix_formatter
             if (row == 0)
                 print_first_cell(symbols.epsilon);
             else
-                print_first_cell(query[row-1]);
-            for(unsigned col = 0; col < matrix.cols(); ++col)
+                print_first_cell(as_string(query[row - 1]));
+            for (size_t col = 0; col < matrix.cols(); ++col)
                 print_cell(entry_at(row, col));
             cout << "\n";
         }
     }
 
 private:
-
     //!\brief Same as #matrix\.at(*row*, *col*), but converts the value to
     //!       a trace symbol (alignment_matrix_format::trace_dir) if the
     //!       #matrix is a traceback matrix.
-    std::string entry_at(unsigned row, unsigned col) const noexcept
+    std::string entry_at(size_t const row, size_t const col) const noexcept
     {
         if constexpr(is_traceback_matrix)
         {
             trace_directions direction = matrix.at(row, col);
-            return std::string{symbols.trace_dir[(unsigned)(direction) % 8u]};
-        } else
+            return symbols.trace_dir[(size_t)(direction) % 8u];
+        }
+        else
         {
-            return as_string(matrix.at(row, col));
+            entry_type entry = matrix.at(row, col);
+            return as_string(entry);
         }
     }
 
     //!\brief Convert a matrix entry into a std::string
-    std::string as_string(auto && entry) const noexcept
+    static std::string as_string(auto && entry) noexcept
     {
         std::stringstream stream;
-        if constexpr (alphabet_concept<remove_cvref_t<decltype(entry)>> &&
-                      !ostream_concept<std::stringstream, decltype(entry)>)
-        {
+        if constexpr (alphabet_concept<remove_cvref_t<decltype(entry)>>)
             stream << to_char(entry);
```

Comment:
@h-2 Can I use the debugstream here, which overloads stringstream?

Reply:
yes